src/script.rs

```diff
@@ -0,0 +1,642 @@
+                max_num_size,
+                actual: vch.len(),
+            });
+        }
+        if require_minimal && !vch.is_empty() {
+            // Check that the number is encoded with the minimum possible
+            // number of bytes.
+            //
+            // If the most-significant-byte - excluding the sign bit - is zero
+            // then we're not minimal. Note how this test also rejects the
+            // negative-zero encoding, 0x80.
+            if (vch.last().unwrap_or_else(|| unreachable!()) & 0x7F) == 0 {
+                // One exception: if there's more than one byte and the most
+                // significant bit of the second-most-significant-byte is set
+                // it would conflict with the sign bit. An example of this case
+                // is +-255, which encode to 0xff00 and 0xff80 respectively.
+                // (big-endian).
+                if vch.len() <= 1 {
+                    return Err(ScriptNumError::NegativeZero);
+                } else if (vch[vch.len() - 2] & 0x80) == 0 {
+                    return Err(ScriptNumError::NonMinimalEncoding);
+                }
+            }
+        }
+        Self::set_vch(vch).map(ScriptNum)
+    }
+
+    pub fn getint(&self) -> i32 {
+        if self.0 > i32::MAX.into() {
+            i32::MAX
+        } else if self.0 < i32::MIN.into() {
+            i32::MIN
+        } else {
+            self.0.try_into().unwrap()
+        }
+    }
+
+    pub fn getvch(&self) -> Vec<u8> {
+        Self::serialize(&self.0)
+    }
+
+    pub fn serialize(value: &i64) -> Vec<u8> {
+        if *value == 0 {
+            return Vec::new();
+        }
+
+        if *value == i64::MIN {
+            // The code below is buggy, and produces the "wrong" result for
+            // INT64_MIN. To avoid undefined behavior while attempting to
+            // negate a value of INT64_MIN, we intentionally return the result
+            // that the code below would produce on an x86_64 system.
+            return vec![0, 0, 0, 0, 0, 0, 0, 128, 128];
+        }
+
+        let mut result = Vec::new();
+        let neg = *value < 0;
+        let mut absvalue = value.abs();
+
+        while absvalue != 0 {
+            result.push(
+                (absvalue & 0xff)
+                    .try_into()
+                    .unwrap_or_else(|_| unreachable!()),
+            );
+            absvalue >>= 8;
+        }
+
+        //    - If the most significant byte is >= 0x80 and the value is positive, push a
+        //    new zero-byte to make the significant byte < 0x80 again.
+
+        //    - If the most significant byte is >= 0x80 and the value is negative, push a
+        //    new 0x80 byte that will be popped off when converting to an integral.
+
+        //    - If the most significant byte is < 0x80 and the value is negative, add
+        //    0x80 to it, since it will be subtracted and interpreted as a negative when
+        //    converting to an integral.
```

Comment:
Fix the weird formatting.
```suggestion
        // - If the most significant byte is >= 0x80 and the value is positive, push a
        //   new zero-byte to make the significant byte < 0x80 again.
        // - If the most significant byte is >= 0x80 and the value is negative, push a
        //   new 0x80 byte that will be popped off when converting to an integral.
        // - If the most significant byte is < 0x80 and the value is negative, add 0x80
        //   to it, since it will be subtracted and interpreted as a negative when
        //   converting to an integral.